src/app/AlwaysOn.Shared/TelemetryExtensions/AppInsightsRequestHandler.cs

```diff
@@ -0,0 +1,80 @@
+﻿using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.Azure.Cosmos;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace AlwaysOn.Shared.TelemetryExtensions
+{
+    public class AppInsightsRequestHandler : RequestHandler
+    {
+        public AppInsightsRequestHandler(TelemetryClient telemetryClient)
+        {
+            _telemetryClient = telemetryClient;
+        }
+
+        private readonly TelemetryClient _telemetryClient;
+
+        public override async Task<ResponseMessage> SendAsync(RequestMessage request, CancellationToken cancellationToken)
+        {
+            var title = request.Properties.ContainsKey("Operation") ? request.Properties["Operation"].ToString() : $"{request.Method} {request.RequestUri.OriginalString}";
+            var dbClientEndpoint = request.Properties.ContainsKey("DbClientEndpoint") ? request.Properties["DbClientEndpoint"].ToString() : null;
+
+            using var dependency = _telemetryClient.StartOperation<DependencyTelemetry>(title);
+
+            //
+            // Comparison with the original telemetry object:
+            //
+            //var telemetry = new DependencyTelemetry()
+            //{
+            //    OK - Type = AppInsightsDependencyType,
+            //    OK - Data = $"ObjectId={objectId}, Partitionkey={partitionKey}",
+            //    OK - Name = $"Delete {typeof(T).Name}",
+            //    OK - Timestamp = startTime,
+            //    OK - Duration = diagnostics != null ? diagnostics.GetClientElapsedTime() : overallDuration,
+            //    OK - Target = diagnostics != null ? diagnostics.GetContactedRegions().FirstOrDefault().uri?.Host : _dbClient.Endpoint.Host,
+            //    REMOVED - Success = success
+            //};
+            //
+            //if (response != null)
+            //    OK - telemetry.Metrics.Add("CosmosDbRequestUnits", response.RequestCharge);
+
+            var response = await base.SendAsync(request, cancellationToken);
+
+            dependency.Telemetry.Type = "Azure DocumentDB";
```

Comment:
wasnt there a constant for this before somewhere? or you got rid of that intentionally?

Reply:
There was, but now this is the only place where this is used, so there's no real need to have a constant for it.